tests/textcase/lsp/lsp_spec.lua

```diff
@@ -1 +1,62 @@
-describe("LSP", function() end)
+local test_helpers = require("tests.test_helpers")
+local textcase = require("textcase")
+
+local cur_dir = vim.fn.expand("%:p:h")
+
+---@param path string
+function M.read_file(path)
+  local fd = vim.loop.fs_open(path, "r", 438)
+  local fstat = vim.loop.fs_fstat(fd)
+  local contents = vim.loop.fs_read(fd, fstat.size, 0)
+  vim.loop.fs_close(fd)
+  return contents
+end
+
+function M.waitFor(max_time, callback)
+  local max_seconds = max_time
+  local curr_seconds = 0
+  local task_finished = false
+  while curr_seconds < max_seconds and not task_finished do
+    curr_seconds = curr_seconds + 0.5
```

Comment:
Nitpick: I would probably assign `local step_time = 0.5` outside the function and use it here as well as in `vim.wait(step_time * 1000, function() end)`.

Reply:
And I would probably try it with 0.1 seconds to speed up the tests.

Reply:
Fair enough